core/src/main/java/ar/com/wolox/wolmo/core/permissions/PermissionManager.kt

```diff
@@ -0,0 +1,123 @@
+package ar.com.wolox.wolmo.core.permissions
+
+import android.content.pm.PackageManager
+import androidx.activity.ComponentActivity
+import androidx.activity.result.contract.ActivityResultContracts
+import androidx.core.app.ActivityCompat.shouldShowRequestPermissionRationale
+import androidx.core.content.ContextCompat
+
+class PermissionManager(
+    private val componentActivity: ComponentActivity,
+    private val permissionsRequested: Map<String, PermissionResponse>
+) {
+    fun request() {
+        val permissionsToRequest = mutableMapOf<String, PermissionResponse>()
+        permissionsRequested.forEach { (permissionName, permissionResponseHandler) ->
+            when {
+                ContextCompat.checkSelfPermission(
+                    componentActivity,
+                    permissionName
+                ) == PackageManager.PERMISSION_GRANTED -> {
+                    permissionResponseHandler.onAlreadyGranted()
+                }
+                shouldShowRequestPermissionRationale(componentActivity, permissionName) -> {
+                    /*
```

Comment:
I would put this comment at the beginning, before declaring the class, what do you think?